packages/themes/dekode-theme/src/custom-variables.css

```diff
@@ -1 +1,14 @@
-@custom-media --theme-desktop-menu (width > 900px);
+@custom-media --theme-viewport-mobile (width > 480px);
+@custom-media --theme-viewport-mobile-only (width <= 480px);
+@custom-media --theme-viewport-mobile-down (width <= 480px);
+@custom-media --theme-viewport-small (width > 600px);
+@custom-media --theme-viewport-small-only (width <= 600px);
+@custom-media --theme-viewport-small-down (width <= 600px);
+@custom-media --theme-viewport-medium (width > 782px);
+@custom-media --theme-viewport-medium-down (width <= 782px);
+@custom-media --theme-viewport-large (width > 960px);
+@custom-media --theme-viewport-large-down (width <= 960px);
+@custom-media --theme-viewport-wide (width > 1280px);
+@custom-media --theme-viewport-wide-down (width <= 1280px);
+@custom-media --theme-viewport-huge (width > 1440px);
```

Comment:
There is a gap between huge and wide. Also, it would be better from my perspective to use >= for the `size` and < for `size-down` (like under the `size` kind of logic)

Reply:
In my head there should not be too many breakpoints:

```
/* Plugin styles */

/* Media queries descriptions.
 *
 *<------- 600px] ------- 960px] ------- 1280px] -------------->
 *
 *
 *<-- small ----|
 *              | -- medium ---|
 *              | -- medium-up -------------------------------->
 *<-- medium-down -------------|
 *                             | -- desktop ---|
 *<-- desktop-down ----------------------------|
 *                             |-- desktop-up ----------------->
 *                                             | -- default -- >
 *
 */

/*
0 -> 600
600 -> 960
960 -> 1280
1280 -> ∞
*/


@custom-media --view-small (width <= 600px);
@custom-media --view-wp-small-down (width <= 780px);

@custom-media --view-medium-down (width < 960px);
@custom-media --view-medium (600px < width < 960px);
@custom-media --view-medium-up (width > 960px);

@custom-media --view-desktop-down (width < 1280px);
@custom-media --view-desktop (960px <= width < 1280px);
@custom-media --view-desktop-up (width >= 960px);

@custom-media --view-mobile-menu (width <= 1280px);

@custom-media --view-default (width >= 1280px);
@custom-media --view-default-up (1280px <= width < 1400px);
@custom-media --view-huge (width >= 1400px);
```

Reply:
Are these still the same breakpoints Gutenberg/WP uses ?

Reply:
I agree with @Sisanu regarding `>=`
e.g.
```css
@custom-media --promon-viewport-medium (width >= 640px);
@custom-media --promon-viewport-medium-down (width < 640px);
`

Reply:
Medium only you might use to fine-tune some of the blocks that need to behave differently in tablet view perhaps? For example, in Digital Norway there is one card with a wave that needs to show the vertical wave up to the point but the card size itself needs to change according to the grid behavior, so you would need to have two breakpoints. It's just an example, and I know this was not designed correctly in the first place, just mentioning why you would still need two of them.

Reply:
Good point there.
Foundation uses the same style (without the `up` keyword)
https://get.foundation/sites/docs/media-queries.html#the-breakpoint-mixin

Reply:
In that case I vote for something like this:
```css
@custom-media --view-medium (width >= 640px);
@custom-media --view-medium-only (width >= 640px and width < whatever_is_next);
@custom-media --view-medium-down (width < 640px);

Reply:
Also don't need both `--theme-viewport-mobile-only` and `--theme-viewport-mobile-down` that has the same value. Should be one.

Reply:
I don't have a strong opinions either way. @Sisanu Can I reach out to William to book 0.5 hours for you to optimize this pull request based on the input in this thread?

Reply:
That's fine @olethomas. But before we do that, can we also ask/sink with a designer on which is the new trend and what are the expectations from the design perspective going further for the breakpoints? Just wanted to be sure we are all on the same page.

Reply:
@cecilieauthen

Reply:
I may be a bit out of date here, but looking at the default being a high resolution, shouldn't this be the other way around? Start with the smallest viewport, then make minor adjustments as the viewport grows (think "mobile first", except we're being device agnostic so we don't call it that any more :P ). This is what is recommended by MDN as well it seems.

This would remove the need for a lot of styles, and keeping in line with search engines approach of asking taht you not differ the content that is available regardless of viewport, we want to avoid viewport-centric sizes that lock elements into just one size (this means singular rules like `medium` and `desktop` in the sheet above are not ideal).

Reply:
For backward compatibility, this will complicate things I guess, but transitioning from PB1 to PB2 will anyway require hardcoding the variables at the theme level, including the default view. Also, currently, the blocks and everything are built the other way around, and it will require the designs to also be changed to accommodate this for all the components. I guess that inverting this will require lots of shifts in the way we think and build everything.

Reply:
The switch between project base setups should not have any impact here, this is a proposal for the new base theme, and only things developed for the new base theme would matter.

We should, however, include the designers in this discussion, as you say, there is a shift in thinking if we do follow the mobile first approach.